### DISCUSSION TITLE
To get lowest energy conformer of a larger molecule using rdkit and/or with GPU acceleration

### DISCUSSION THREAD

**Discussion by rajarshiche** [rdkit/rdkit#5139] [General]
Is there a way to get the lowest energy conformer of a bigger SMILES string (MW ~ 2400) interfacing rdkit with other libraries ?
(SMILES e.g. 'CC(=O)N[C@@H](Cc1ccccc1)C(=O)N[C@@H](CCC(N)=O)C(=O)N[C@@H(CO)C(=O)N[C@@H](CCCCN)C(=O)N[C@H](C(=O)N[C@@H(C)C(=O)N[C@H](C(=O)N[C@@H](Cc1ccccc1)C(=O)N[C@H](C(=O)N[C@@H(CC(=O)O)C(=O)NCC(=O)N[C@@H](Cc1ccc(O)cc1)C(=O)N[C@@H(Cc1ccccc1)C(=O)N[C@@H](CCC(=O)O)C(=O)N[C@@H(CCCNC(=N)N)C(=O)N[C@@H](CC(C)C)C(=O)N[C@@H(CS)C(=O)N[C@@H](C)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CC(C)C)C(N)=O)C(C)C)C(C)C)C(C)C')

I have tried AllChem.MMFFGetMoleculeOpenMMForceField function using rdkit, but could not make it to work. Also missing an appropriate and working use case of gpu accelerated conformer generation through AllChem.MMFFGetMoleculeOpenMMForceField function using rdkit. 

Any help and guidance would be appreciated here.

**Comment by greglandrum** [comment]
The RDKit has no GPU integration.
If you want help with the other things, you need to provide us with specific examples of the code you used and the errors you got.

**Reply by rajarshiche** [reply]
Hi Greg, Thanks for the help here. I am sharing a simple conformer generation code in the following link:

https://github.com/rajarshiche/Conformer-generation/blob/main/Confomer_generation_MMFF94s.py

There are 2 problems here: (1) For the given smile strings in the above link, always the error comes as 'Bad Conformer Id' and (2) when a different and larger smile string (MW ~ 2500) is given as input, it simply takes multiple hours to execute. An example of such smile string is 'CC(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)NCC(=O)N[C@@H](CCCNC(=N)N)C(=O)N[C@@H](CC(C)C)C(=O)N[C@@H](CCC(N)=O)C(=O)N[C@H](C(=O)N[C@@H](C)C(=O)N[C@@H](Cc1ccccc1)C(=O)N[C@@H](CCC(N)=O)C(=O)N[C@@H](CO)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CC(N)=O)C(=O)N[C@H](C(=O)N[C@@H](Cc1ccccc1)C(=O)N[C@H](C(=O)N[C@@H](Cc1ccccc1)C(=O)NCC(=O)N[C@@H](Cc1ccc(O)cc1)C(=O)N[C@@H](Cc1ccccc1)C(=O)N[C@@H](CCC(=O)O)C(=O)N[C@@H](CCCNC(=N)N)C(=O)N[C@@H](CC(C)C)C(=O)N[C@@H](CCCNC(=N)N)C(=O)N[C@@H](C)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CC(C)C)C(=O)N[C@H](C(=O)N[C@@H](CO)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(N)=O)[C@@H](C)O)C(C)C)C(C)C)C(C)C'

How to get rid of 'Bad Conformer Id' type errors and make the script run faster for larger smiles strings?
